pywren/pywren_ibm_cloud/wren.py

```diff
@@ -322,8 +360,22 @@ def get_result(self, futures=None, throw_except=True, timeout=wrenconfig.CF_RUNT
         if(logger.getEffectiveLevel() == logging.WARNING and self.data_cleaner):
             print(msg)
 
-        if result and len(result) == 1:
-            return result[0]
+        statuses = [f.run_status for f in ftrs]
+
+        if result is not None:
+            if len(result) == 0:
```

Comment:
@omerb01 Is it possible `len(result) = 0`? In what circumstances did you get this case?

Reply:
but `wait()` method in `get_result()` waits forever for results till they are available: https://github.com/pywren/pywren-ibm-cloud/blob/7e75e6336db8351e3d8889f9c053087b139fa27c/pywren/pywren_ibm_cloud/wren.py#L320)
so I think this case will never happen, since the list will never be empty.

Reply:
cause of pywren doesn't know by the ID if invocations done or not (gets `ftrs` list empty from storage), I disabled the wait bar only when using pywren ID and managed it by a warning msg instead.

Reply:
the logic behind is that `get_result()` creates dummy-futures after getting calls ids from storage so PyWren can't know if results havne't done yet or if the given ID maybe wrong. This is the reason why I disabled the wait bar when using pywren ID

Reply:
ah ok! I got confused by this logic at the end of the `get_result()` metod. It might be easier to track this logic by printing the message (and return None) immediately after this line (if ftrs list is empty):
https://github.com/pywren/pywren-ibm-cloud/blob/7e75e6336db8351e3d8889f9c053087b139fa27c/pywren/pywren_ibm_cloud/wren.py#L288

then, this line wouldn't be necessary, since timeout and bar are also useful if PyWrenID is provided:
https://github.com/pywren/pywren-ibm-cloud/blob/7e75e6336db8351e3d8889f9c053087b139fa27c/pywren/pywren_ibm_cloud/wren.py#L307

Reply:
ok thanks 👍 will change it